eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/EventDataBatchTest.java

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.implementation.logging.ServiceLogger;
+import com.azure.eventhubs.implementation.ApiTestBase;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.util.Random;
+import java.util.UUID;
+
+public class EventDataBatchTest extends ApiTestBase {
+    ServiceLogger logger = new ServiceLogger(EventDataBatchTest.class);
+
+    private static final String CONSUMER_GROUP_NAME = ApiTestBase.getConsumerGroupName();
+    private static final String PARTITION_ID = "0";
+    private static final String PARTITION_KEY = "key1";
+    private static final String APPLICATION_PROPERTY = "firstProp";
+    private static final String MESSAGE_ANNOTATION = "message-annotation-1";
+    private static final String PAYLOAD = "testmsg";
+
+    private static EventHubClient ehClient;
+    private static EventSender sender;
+    private static EventReceiver receiver;
+
+    @Rule
+    public TestName testName = new TestName();
+
+    @Override
+    protected String testName() {
+        return testName.getMethodName();
+    }
+
+    @BeforeClass
+    public static void initialize() {
+        ehClient = ApiTestBase.getEventHubClientBuilder().build();
+
+        EventSenderOptions senderOptions = new EventSenderOptions().partitionId(PARTITION_ID);
+        sender = ehClient.createSender(senderOptions);
+
+        EventReceiverOptions receiverOptions = new EventReceiverOptions()
+            .consumerGroup(ApiTestBase.getConsumerGroupName())
+            .beginReceivingAt(EventPosition.newEventsOnly());
+        receiver = ehClient.createReceiver(PARTITION_ID, receiverOptions);
+    }
+
+    @AfterClass
+    public static void cleanupClient() {
+        if (ehClient != null) {
+            ehClient.close();
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void nullEventData() {
+        final EventDataBatch batch = new EventDataBatch(1, PARTITION_KEY);
+        batch.tryAdd(null);
+    }
+
+    @Test(expected = AmqpException.class)
```

Comment:
Can we assert more than this? That the AmqpException contains the correct ErrorCondition.

Reply:
Remove the `expected =`. Update the test to:

```java
// Arrange 
final EventDataBatch batch = new EventDataBatch(1024, PARTITION_KEY); 
final EventData tooBig = new EventData(new byte[1024 * 1024 * 2]); 

// Act & Assert
try {
    batch.tryAdd(tooBig);
    Assert.fail("Expected an exception");
} catch (AmqpException e) {
    Assert.assertFalse(e.isTransient());
    Assert.assertEquals(ErrorCondition.LINK_PAYLOAD_SIZE_EXCEEDED, e.getErrorCondition());
}
```

Reply:
Thanks.  Better than what I have coming up.